### DISCUSSION TITLE
Run functions at different rates in Python API for Crazyflie

### DISCUSSION THREAD

**Discussion by Allanadn** [bitcraze/discussions#976] [Q&A]
Dear Crazyflie team,

I am new to the Crazyflie environment and Python, so I thank you in advance for your patience and for bearing with me. I am currently using the Python API for Crazyflie 2.1 as a wrapper to make it take off and go to desired positions with the "go_to" method in the "PositionHlCommander". I have structured the code so that I have a motion planner in Matlab which should report a (x,y) positions that the UAV should follow. This function should run and report new positions for the crazyflie to follow each 0.2s. I link the motion planner with the Crazyflie through the Matlab engine for Python and the expect the UAV to follow these positions. In the code I end up calling this function every time I get the new position from the Kalman estimator (I use the FlowDeck V2). Is there a simple way I could make this function run every 0.2s and when no new information from this Matlab function is available, just keep using the old info?  

P.s.: As for now I am using the time.sleep(0.2) command straight after the Motion Planner to force the behavior, although I am not sure this is working...

Thank you very much for your time and attention again

**Comment by danielcmdeveloper** [comment]
Good afternoon,
We are also trying to do that, did you finally manage to solve it? If so, could you tell us how?
Greetings and thank you very much

**Reply by Allanadn** [reply]
Hi daniel, unfortunately I did not find any neat way of performing this task. Please let me know if you find something.
Cheers!

**Comment by danielcmdeveloper** [comment]
Good afternoon,
Understood, when I have the solution I will report it here so I can help you.
Thank you very much for the answer.
We will keep in touch.
Regards

**Comment by knmcguire** [comment]
@Allanadn this seems to be closed. Have you found a solution and if so, could you add that here and mark as an answer?
